Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import CoreData
+import AutomatticTracks
+
+@objc
+class SharedStorageMigrator: NSObject {
+    private let storageSettings: StorageSettings
+    private let fileManager: FileManager
+
+    init(storageSettings: StorageSettings = StorageSettings(), fileManager: FileManager = FileManager.default) {
+        self.storageSettings = storageSettings
+        self.fileManager = fileManager
+    }
+
+    var legacyStorageExists: Bool {
+        fileManager.fileExists(atPath: storageSettings.legacyStorageURL.path)
+    }
+
+    var sharedStorageExists: Bool {
+        fileManager.fileExists(atPath: storageSettings.sharedStorageURL.path)
+    }
+
+    /// Database Migration
+    /// To be able to share data with app extensions, the CoreData database needs to be migrated to an app group
+    /// Must run before Simperium is setup
+
+    func performMigrationIfNeeded() -> MigrationResult {
+        // Confirm if the app group DB exists
+        guard migrationNeeded else {
+            NSLog("Core Data Migration not required")
+            return .notNeeded
+        }
+
+        return migrateCoreDataToAppGroup()
+    }
+
+    private var migrationNeeded: Bool {
+        return legacyStorageExists && !sharedStorageExists
+    }
+
+    private func migrateCoreDataToAppGroup() -> MigrationResult {
+        // Testing prints
+        // TODO: Remove prints later
+        print("oldDb exists \(FileManager.default.fileExists(atPath: storageSettings.legacyStorageURL.path))")
+        print(storageSettings.legacyStorageURL.path)
+        print("newDb exists \(FileManager.default.fileExists(atPath: storageSettings.sharedStorageURL.path))")
+        print(storageSettings.sharedStorageURL.path)
+
+        NSLog("Database needs migration to app group")
+        NSLog("Beginning database migration")
+
+        do {
+            try disableJournaling()
+            try migrateCoreDataFiles()
+            NSLog("Database migration successful!!")
+            return .success
+        } catch {
+            NSLog("Could not migrate database to app group")
+            NSLog(error.localizedDescription)
+            CrashLogging.logError(error)
+            return .failed
+        }
+    }
+
+    private func disableJournaling() throws {
```

Comment:
Addendum: can we please add a link to [this Apple TechNote](https://developer.apple.com/library/archive/qa/qa1809/_index.html) for further reference?

(Thank you!!)